ui/src/alerting/components/notifications/EditRuleOverlay.tsx

```diff
@@ -0,0 +1,114 @@
+  Input,
+  Grid,
+  Columns,
+} from '@influxdata/clockface'
+
+// Reducers
+import {reducer, ActionPayload} from './RuleOverlay.reducer'
+
+// Constants
+import {endpoints} from 'src/alerting/constants' // Hooks
+import {RuleModeContext, EditRuleDispatch, RuleMode} from 'src/shared/hooks'
+
+// Types
+import {NotificationRuleDraft, AppState} from 'src/types'
+
+interface StateProps {
+  stateRule: NotificationRuleDraft
+}
+
+type Props = WithRouterProps & StateProps
+
+const EditRuleOverlay: FC<Props> = ({params, router, stateRule}) => {
+  if (!stateRule) {
+    return null
+  }
+
+  const handleDismiss = () => {
+    router.push(`/orgs/${params.orgID}/alerting`)
+  }
+
+  const mode = RuleMode.Edit
+  const [rule, dispatch] = useReducer(reducer(mode), stateRule)
+  const ruleDispatch = (action: ActionPayload): void => {
+    dispatch({...action, mode})
+  }
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const {name, value} = e.target
+    dispatch({
+      type: 'UPDATE_RULE',
+      mode,
+      rule: {...rule, [name]: value} as NotificationRuleDraft,
+    })
+  }
+
+  return (
+    <RuleModeContext.Provider value={mode}>
+      <EditRuleDispatch.Provider value={ruleDispatch}>
+        <Overlay visible={true}>
+          <Overlay.Container maxWidth={800}>
+            <Overlay.Header
+              title="Edit this Notification Rule"
+              onDismiss={handleDismiss}
+            />
+            <Overlay.Body>
+              <Grid>
+                <Form>
+                  <Grid.Row>
+                    <Grid.Column widthSM={Columns.Two}>About</Grid.Column>
+                    <Grid.Column widthSM={Columns.Ten}>
+                      <Panel size={ComponentSize.ExtraSmall}>
+                        <Panel.Body>
+                          <Form.Element label="Name">
+                            <Input
+                              testID="rule-name--input"
+                              placeholder="Name this new rule"
+                              value={rule.name}
+                              name="name"
+                              onChange={handleChange}
+                            />
+                          </Form.Element>
+                          <RuleSchedule rule={rule} onChange={handleChange} />
+                        </Panel.Body>
+                      </Panel>
+                    </Grid.Column>
+                    <Grid.Column>
+                      <hr />
+                    </Grid.Column>
+                  </Grid.Row>
+                  <RuleConditions rule={rule} />
+                  <RuleMessage rule={rule} endpoints={endpoints} />
+                </Form>
+              </Grid>
+            </Overlay.Body>
+          </Overlay.Container>
+        </Overlay>
+      </EditRuleDispatch.Provider>
+    </RuleModeContext.Provider>
+  )
+}
+
+const mstp = ({rules}: AppState, {params}: Props): StateProps => {
+  const stateRule = rules.list.find(r => r.id === params.ruleID)
```

Comment:
A cool thing about hooks I realized lately is that they provide really natural ways to write memoized redux selectors.

Instead of doing [this really awkward annoying thing](https://github.com/reduxjs/reselect#sharing-selectors-with-props-across-multiple-component-instances), you can pass the dependencies of the selector to the component and use `useMemo` instead. Seems nicer to me.

(Not relevant here, I don't think this needs a selector.)